lib/services/lwm2mHandlers/registration.js

```diff
@@ -188,10 +188,20 @@ function addUnsupportedAttributes(payload, configuration, deviceInformation, cal
             if (omaRegistry[value.objectType].resources) {
                 return previous.concat(omaRegistry[value.objectType].resources);
             } else {
-                previous.push({
-                    name: omaRegistry[value.objectType].name + '#' + value.objectInstance,
-                    type: 'string'
-                });
+
+                // In NGSIv2, spaces cannot be used in attributes' names. Therefore, we trim them.
```

Comment:
Why this problem has arised when developing the NGSIv2-based provisioning but not in the previous NGSIv2-based active attributes values sending?

Or the situation was the same and also the same the solution applied (i.e. trim spaces)?

Reply:
Because there is not any test as part of iotagent-node-lib, iotagent-ul or iotagent-json that contains spaces in the attributes' names... Maybe it is something that we must solve at iotagent-node-lib level?

Reply:
If this something common to all the agents, iotagent-node-lib seems a good place to cover it with tests.

Reply:
I have been doing some tests regarding this issue and it is more complicated than it seemed initially.

If we decide that iotagent-node-lib will escape spaces in device's attributes automatically, this change may imply a big amount of work since it affects to device provisioning, forwarding of active attributes, plugins... 

In my eyes, the best solution is:
- At iotagen-node-lib level, to follow the same approach as in Orion CB and reject device provisioning requests that attributes' names with spaces.
- At lightweightm2m-iotagent, since https://github.com/telefonicaid/lightweightm2m-iotagent/blob/master/omaRegistry.json contains a lot of spaces, we could implement space escaping.

Reply:
Just to check if I'm understanding correctly...

You mean not touching iotagent-node-lib and implement space adaptation in the agent code itself? (either removing it as the PR shows now or apply %20 escaping). Is that correct?

Reply:
Yes, that's correct. I agree with you to implementing %20 escaping but just in this specific IOTA. Implementing this in the iotagent-node-lib would be a very big effort and the benefit is very small...

Reply:
I agree. So the idea is to change the current mechanism in this PR (removing spaces) to the new one (escaping spaces using %20)?

Reply:
Yes, I will implement this ASAP. :)

Reply:
Fixed in 9eadfbbd69357aef810b9820f8c12e1a26b8614b